module/vision/GoalDetector/src/GoalDetector.cpp

```diff
@@ -398,11 +542,11 @@ namespace vision {
                     arma::vec2 br = imageToScreen(quad.getBottomRight(), convert<uint, 2>(image.dimensions));
                     arma::vec2 screenGoalCentre = (tl + tr + bl + br) * 0.25;
 
-                    // Get vectors for TL TR BL BR goal;
-                    arma::vec3 ctl       = getCamFromScreen(tl, cam.focalLengthPixels);
-                    arma::vec3 ctr       = getCamFromScreen(tr, cam.focalLengthPixels);
-                    arma::vec3 cbl       = getCamFromScreen(bl, cam.focalLengthPixels);
-                    arma::vec3 cbr       = getCamFromScreen(br, cam.focalLengthPixels);
+                    // Get vectors for TL TR BL BR;
+                    arma::vec3 ctl       = convert<double, 3>(it->frustum.tl);
+                    arma::vec3 ctr       = convert<double, 3>(it->frustum.tr);
+                    arma::vec3 cbl       = convert<double, 3>(it->frustum.bl);
+                    arma::vec3 cbr       = convert<double, 3>(it->frustum.br);
                     arma::vec3 rGCc_norm = arma::normalise((cbl + cbr) * 0.5);  // vector to bottom centre of goal post
 
                     // TODO: NORMALS not used currently - delete?
```

Comment:
Delete this todo, the corners are good enough to calculate the normals if needed